processing-service-core/api/models/db/repositoryConfigStore.js

```diff
@@ -0,0 +1,186 @@
+                " repositoryId '" +repositoryId+ "', key '" +key+"', and tenantId '"+ tenantId +"'."));
+        });
+
+    return deferredDelete.promise;
+}
+
+/**
+ * Retrieves the value for a repository config matching the specified parameters.
+ * @param repositoryId {String} the repository ID identifying the tenant config.
+ * @param tenanatId {String} the tenant ID identifying the config that should be deleted.
+ * @param key {String} the key identifying the repository config.
+ * @returns {*|d.promise|Function|promise|a|h} a promise that will be resolved or rejected based on the result of retrieval.
+ *  Resolved promise will return the value on the matched tenant config.
+ *  Rejected promise can pass an ApiError with type set to either DATABASE_UNKNOWN_ERROR or ITEM_NOT_FOUND if unable to find
+ *  the specified tenant config.
+ */
+function getRepositoryConfig(repositoryId, tenantId, key) {
+    var deferredGet = Q.defer();
+
+    repositoryConfigTable.findOne({
+            attributes: [tableDefinition.value.field],
+            where: { repositoryId: repositoryId, key: key, tenantId: tenantId}
+        })
+        .then(function(findResult) {
+            if(findResult===null) {
+                logger.error("Failed to find repository config with repositoryId '"+repositoryId+"', key '"+key+"', and tenantId '"+ tenantId +"'.");
+                deferredGet.reject(apiErrorFactory.createNotFoundError(
+                    "Failed to find repository config with repositoryId '"+repositoryId+"', key '"+key+"', and tenantId '"+ tenantId +"'."));
+                return;
+            }
+            deferredGet.resolve(findResult.value);
+        })
+        .catch(function(errorResponse) {
+            logger.error("Failure occurred trying to get repository config with repositoryId '"+repositoryId+"', key '"+key+"', and tenantId '"+ tenantId +"': " + errorResponse.toString());
+            deferredGet.reject(apiErrorFactory.createDatabaseUnknownError(
+                "Failure during retrieval of tenant config with repositoryId '"+repositoryId+"', key '"+key+"', and tenantId '"+ tenantId +"'."));
+        })
+        .done();
+
+    return deferredGet.promise;
+}
+
+/**
+ * Retrieves existing repository configs for the specified tenant and repository. The first 100 repository configs will be returned.
+ * @param repositoryId {String} the tenant ID to return tenant configs for.
+ * @param tenanatId {String} the tenant ID identifying the config that should be deleted.
+ * @returns {*|d.promise|Function|promise|a|h} a promise that will be resolved or rejected based on the result of retrieval.
+ *  Resolved promise will return an array of containing the keys and values for the repository configs matched by the tenant ID and repository ID.
+ *  e.g. [ { 'key': entity', 'value': 'ip_address' } ]
+ *  Rejected promise can pass an ApiError with type set to either DATABASE_UNKNOWN_ERROR.
+ */
```

Comment:
Review function doc (similar issues as the last one)
(and same for next method)